spec/overcommit/hook_context/pre_commit_spec.rb

```diff
@@ -7,6 +7,41 @@
   let(:input) { double('input') }
   let(:context) { described_class.new(config, args, input) }
 
+  describe '#amend?' do
+    subject { context.amend? }
+
+    context 'when amending a commit using `git commit --amend`' do
+      before do
+        Overcommit::Utils.stub(:parent_command).and_return('git commit --amend')
+      end
+
+      it { should == true }
+    end
+
+    context 'when amending a commit using a git alias' do
+      around do |example|
+        repo do
+          `git config alias.amend 'commit --amend'`
+          example.run
+        end
+      end
+
+      before do
+        Overcommit::Utils.stub(:parent_command).and_return('git amend')
+      end
+
+      it { should == true }
+    end
```

Comment:
Should also add a quick test ensuring we don't match silly aliases like:

```ini
[alias]
silly--amend = commit --amend
```

...which _is_ possible if you see my comment above about hypens in aliases.